govcd/api.go

```diff
@@ -178,3 +178,100 @@ func checkResp(resp *http.Response, err error) (*http.Response, error) {
 		return nil, fmt.Errorf("unhandled API response, please report this issue, status code: %s", resp.Status)
 	}
 }
+
+// Helper function creates request, run it, check response and parse task from response.
+// pathUlr - request URL
+// requestType - HTTP method type
+// contentType - value which will be set for "Content-Type"
+// errorMessage - error message which to return when error happen
+// payload - XML struct which will be marshalled and added as body/payload
+// E.g. client.ExecuteTaskRequest(updateDiskLink.HREF, http.MethodPut, updateDiskLink.Type, "error update disk: %s", xmlPayload)
+func (client *Client) ExecuteTaskRequest(pathURL, requestType, contentType, errorMessage string, payload interface{}) (Task, error) {
+
+	resp, err := executeRequest(pathURL, requestType, contentType, errorMessage, payload, client)
+	if err != nil {
+		return Task{}, fmt.Errorf(errorMessage, err)
+	}
+
+	task := NewTask(client)
+
+	if err = decodeBody(resp, task.Task); err != nil {
+		return Task{}, fmt.Errorf("error decoding Task response: %s", err)
+	}
+
+	resp.Body.Close()
```

Comment:
Let's check the return error.

In general, when we decide to replace many copy-and-paste with a single dynamic function, that function needs to be as powerful as possible, and also have all the checks that would look too verbose for repeated code.

Reply:
Fixed